core/trino-main/src/main/java/io/trino/operator/scalar/timestamp/VarcharToTimestampCast.java

```diff
@@ -85,16 +86,22 @@ public static long castToShortTimestamp(int precision, String value)
         String second = matcher.group("second");
         String fraction = matcher.group("fraction");
 
-        long epochSecond = ZonedDateTime.of(
-                Integer.parseInt(year),
-                Integer.parseInt(month),
-                Integer.parseInt(day),
-                hour == null ? 0 : Integer.parseInt(hour),
-                minute == null ? 0 : Integer.parseInt(minute),
-                second == null ? 0 : Integer.parseInt(second),
-                0,
-                UTC)
-                .toEpochSecond();
+        long epochSecond;
+        try {
+            epochSecond = ZonedDateTime.of(
+                            Integer.parseInt(year),
+                            Integer.parseInt(month),
+                            Integer.parseInt(day),
+                            hour == null ? 0 : Integer.parseInt(hour),
+                            minute == null ? 0 : Integer.parseInt(minute),
+                            second == null ? 0 : Integer.parseInt(second),
+                            0,
+                            UTC)
+                    .toEpochSecond();
+        }
+        catch (DateTimeException e) {
+            throw new IllegalArgumentException("Invalid timestamp: " + value, e);
```

Comment:
TrinoException

Reply:
Is it better to replace another IllegalArgumentException in this method?
https://github.com/trinodb/trino/blob/102ab2ac89136b02fe745abfa00af50937a0ea31/core/trino-main/src/main/java/io/trino/operator/scalar/timestamp/VarcharToTimestampCast.java#L120-L122

Reply:
Yes, please